game/ringbuffer.hh

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <atomic>
+
+
+/// Lock-free ring buffer. Discards oldest data on overflow (not strictly thread-safe).
+template <std::size_t SIZE> class RingBuffer {
+  public:
+	constexpr static std::size_t capacity = SIZE;
+
+	template <typename InIt> void insert(InIt begin, InIt end);
+	/// Read data from current position if there is enough data to fill the range (otherwise return false). Does not move read pointer.
+	template <typename OutIt> bool read(OutIt begin, OutIt end);
+	void pop(std::size_t n); ///< Move reading pointer forward.
+	std::size_t size() const;
+
+  private:
+	static std::size_t modulo(std::size_t idx);  ///< Modulo operation with proper rounding (handles slightly "negative" idx as well)
+
+  private:
```

Comment:
please do not define double private sections
Do keep the methods together and fields together though

I'd prefer the order of method - member